ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-traffic-engineering.j2

```diff
@@ -0,0 +1,56 @@
+{# eos - Router Traffic Engineering #}
+{% if router_traffic_engineering is arista.avd.defined %}
+!
+router traffic-engineering
+{%     if router_traffic_engineering.segment_routing is arista.avd.defined %}
+{%         set ns = namespace(delimiter=false) %}
+   segment-routing
+{%         if router_traffic_engineering.segment_routing.colored_tunnel_rib is arista.avd.defined(true) %}
+{%             set ns.delimiter = true %}
+      rib system-colored-tunnel-rib
+{%         endif %}
+{%         if router_traffic_engineering.segment_routing.policy_endpoints is arista.avd.defined %}
+{%             for endpoint in router_traffic_engineering.segment_routing.policy_endpoints | arista.avd.natural_sort('address') %}
+{%                 for color in endpoint.colors | arista.avd.natural_sort('value') %}
+{%                     if ns.delimiter is arista.avd.defined(true) %}
+      !
+{%                     endif %}
+{%                     set ns.delimiter = true %}
+      policy endpoint {{ endpoint.address }} color {{ color.value }}
+{%                     if color.binding_sid is arista.avd.defined %}
+         binding-sid {{ color.binding_sid }}
+{%                     endif %}
+{%                     if color.name is arista.avd.defined %}
+         name {{ color.name }}
+{%                     endif %}
+{%                     if color.description is arista.avd.defined %}
+         description {{ color.description }}
+{%                     endif %}
+{%                     for pathgroup in color.path_group | arista.avd.natural_sort('preference') %}
+         !
+         path-group preference {{ pathgroup.preference }}
+{%                         if pathgroup.explicit_null is arista.avd.defined %}
+            explicit-null {{ pathgroup.explicit_null }}
+{%                         endif %}
+{%                         for labelstack in pathgroup.segment_list | arista.avd.natural_sort('index') %}
```

Comment:
The sort key should be ```label_stack``` not index.

FYI: 
<img width="477" alt="image" src="https://user-images.githubusercontent.com/88734745/154189403-41643d75-b703-4460-9561-9216353a2332.png">

Reply:
This is natural_sort, so this can handle the combination of both int and string.